File: packages/audio/android/src/main/java/dev/tcode/thinmpr/audio/model/contract/SongModelContract.kt
package dev.tcode.thinmpr.audio.model.contract

import android.net.Uri
import dev.tcode.thinmpr.audio.model.valueObject.AlbumId
import dev.tcode.thinmpr.audio.model.valueObject.ArtistId
import dev.tcode.thinmpr.audio.model.valueObject.SongId

interface SongModelContract {
    val id: SongId
    val name: String
    val albumId: AlbumId
    val albumName: String
    val artistId: ArtistId
    val artistName: String
    val imageId: String
    val imageUri: Uri
    val mediaUri: Uri
    val duration: Int
    val trackNumber: Double
}

File: packages/audio/ios/Model/Contract/SongModelContract.swift
import MediaPlayer

protocol SongModelContract {
    var media: MPMediaItemCollection { get }

    var id: SongId { get }

    var name: String { get }

    var albumId: AlbumId { get }

    var albumName: String { get }

    var artistId: ArtistId { get }

    var artistName: String { get }

    var imageId: String { get }

    var artwork: MPMediaItemArtwork? { get }

    var duration: Double { get }
    
    var trackNumber: Int { get }
}
